tables/table.py

```diff
@@ -1458,7 +1500,12 @@ class Record(IsDescription):
 
         return self._where(condition, condvars, start, stop, step)
 
-    def _where(self, condition, condvars, start=None, stop=None, step=None):
+    def _where(self,
+               condition: str,
+               condvars: Optional[dict[str, Union["Column", np.ndarray]]],
+               start: Optional[str]=None,
+               stop: Optional[str]=None,
+               step: Optional[str]=None) -> Iterator:
```

Comment:
See the comment on `__iter__`

Reply:
As recommended changed to `Iterator[tableextension.Row]`.